ppa/archive/solr.py

```diff
@@ -28,14 +28,63 @@ class SolrSchema(object):
     '''Solr Schema object.  Includes project schema configuration and
     methods to update configured Solr instance.'''
 
+    # ported from winthrop-django
+    field_types = [
+
+        {
+            'name': 'text_en',
+            "class":"solr.TextField",
+            # for now, configuring index and query analyzers the same
+            # if we want synonyms, query must be separate
+            "analyzer": {
+                # "charFilters": [],
+                "tokenizer": {
+                    "class": "solr.StandardTokenizerFactory",
+                },
+                "filters": [
+                    {"class": "solr.StopFilterFactory"},
+                    {"class": "solr.LowerCaseFilterFactory"},
+                    {"class": "solr.EnglishPossessiveFilterFactory"},
+                    {"class": "solr.KeywordMarkerFilterFactory"},
+                    {"class": "solr.PorterStemFilterFactory"},
+                    {"class": "solr.ICUFoldingFilterFactory"},
+                ]
+            }
+        },
+        {
+            'name': 'string_i',
+            "class": "solr.TextField",
+            "sortMissingLast": True,
+            "analyzer": {
+                "tokenizer": {
+                    # treat entire field as a single token
+                    "class": "solr.KeywordTokenizerFactory",
+                },
+                "filters": [
+                    # lower case to ensure alphabetical sort behaves as
+                    # expected
+                    {"class": "solr.LowerCaseFilterFactory"},
+                    # enable normalization without case folding
+                    # (preserve case for facets)
```

Comment:
If we're not using this field type for facets (which we shouldn't because we don't want them lower case), does this matter? We might be able to use the combined ICU filter that I think we're using for Winthrop text in that case. Not a huge difference, but might make this a little simpler.

Reply:
Of course. I will gladly pull this out.